code/integration/Main/Parser.cpp

```diff
@@ -207,20 +223,31 @@ bool Parser::CheckString(String& OrderString)
   int result = 1;
   String Geom, Version,Arrangement;
   bool Status;
-  
+
+  Serial.print(F("Check string orderstring: ") ); 
+  Serial.println(OrderString);
   //check first string: Geometry
   pos = OrderString.indexOf(";");
   Geom = OrderString.substring(0,pos);
+  Serial.print(F("Check string geom: " )); 
+  Serial.println(Geom);
   if ((Geom.length()<1)or(Geom.length()>2)){result =result *0; }
-  OrderString = OrderString.substring(pos+1);
+  Serial.print(F("Check string first part OK? ") ); 
+  Serial.println(result);
+  OrderString = OrderString.substring(pos+1);                     // ERROR OCCURS HERE ORDERSTRING IS EMPTY after usng substring
+  Serial.print(F("Check string orderstring sub: ") ); 
+  Serial.println(OrderString);
   Geom_Int=Geom.toInt();
+  Serial.print(F("Check string geom int: ") ); 
+  Serial.println(Geom_Int);
   
   //check second string: Version
   pos = OrderString.indexOf(";");
   Version = OrderString.substring(0,pos);
   OrderString = OrderString.substring(pos+1);
-  if (Version.length()!=1) {result =result *0;}
```

Comment:
Die Variable result wird mit 1 initialisiert. Sobald die Länge eines Blockes mit der erwarteten Länge nicht übereinstimmt, wird result=0. Ich habe es nur falsch geschrieben, vielleicht hatte ich erstmal was anderes vor. Das kann man gerne mit result=0 ersetzen :)))) Ich glaube erstmal wollte ich eigentlich zwischen -1 und 1 toggeln.... dann ist das ganze in so einem Zwischenstand geblieben. Sorry :)